src/test/java/seedu/address/logic/commands/UndoCommandTest.java

```diff
@@ -0,0 +1,100 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PET;
+import static seedu.address.testutil.TypicalPets.getTypicalAddressBook;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.pet.Pet;
+import seedu.address.testutil.PetBuilder;
+
+public class UndoCommandTest {
```

Comment:
Would you be adding more test cases, especially for `AttendanceCommand`?

Reply:
Yep will definitely add test cases for the other commands as well. I'll let you know if I face any issues with undoing Attendance thank you! 👍